pkg/controller/gc_controller.go

```diff
@@ -115,6 +115,12 @@ func (c *gcController) cleanBackups() {
 			}
 
 			for _, volumeBackup := range backup.Status.VolumeBackups {
+				if c.snapshotService == nil {
+					glog.Warningf("Cannot garbage-collect snapshot %s for %backup s/%s because server is not configured with PersistentVolumeProvider",
```

Comment:
Maybe let's split out this last commit into a different PR so we can get the config validation in while we continue to discuss the optional PVProvider?